transport/http.go

```diff
@@ -160,7 +160,7 @@ func (c *HTTPTransport) send(spans []*j.Span) error {
 	io.Copy(ioutil.Discard, resp.Body)
```

Comment:
while method & url may be useful, I think this is the place where we're losing the most of the information, as many servers would return some error message in the response, not just the status code.

Reply:
in this case, shouldn't we use the `defer` func here to handle this?

```golang
defer func() {
 io.Copy(ioutil.Discard, resp.Body)
 resp.Body.Close()
}
if resp.StatusCode >= http.StatusBadRequest {
 ...
}
```

Also I added a new test case to handle this.

cc: @necatican

Reply:
No, my point is that we should not be discarding the body. See how sampler handles the response

https://github.com/jaegertracing/jaeger-client-go/blob/7aa7af5fd0410a778e14e0a50d062ce4f7cb9e73/sampler_remote.go#L336-L343

Reply:
+1 to logging the body